test/shared/src/main/scala/zio/test/FailureDetails.scala

```diff
@@ -19,12 +19,13 @@ package zio.test
 /**
  * `FailureDetails` keeps track of details relevant to failures.
  */
-final case class FailureDetails(assertion: ::[AssertionValue], gen: Option[GenFailureDetails] = None) {
```

Comment:
We're losing the static knowledge here that there is at least one failure inside `FailureDetails`. Can we revert this change?

Reply:
Done.
IMHO there should be a custom `NonEmptyList` instead of invariant `::`. As I know in scala 3 it's not a enum branch without its own type only due to compatibility reasons.

Reply:
I agree and there will be a ZIO version of one in ZIO Prelude and `::` is actually covariant in Scala 2.13, but we've got to work with the tools we've got here.